Source/Dafny/Compiler.cs

```diff
@@ -88,18 +91,18 @@ protected interface IClassWriter {
       BlockTargetWriter/*?*/ CreateFunction(string name, List<TypeParameter>/*?*/ typeArgs, List<Formal> formals, Type resultType, Bpl.IToken tok, bool isStatic, bool createBody, MemberDecl member);
       BlockTargetWriter/*?*/ CreateGetter(string name, Type resultType, Bpl.IToken tok, bool isStatic, bool createBody, MemberDecl/*?*/ member);  // returns null iff !createBody
       BlockTargetWriter/*?*/ CreateGetterSetter(string name, Type resultType, Bpl.IToken tok, bool isStatic, bool createBody, MemberDecl/*?*/ member, out TargetWriter setterWriter);  // if createBody, then result and setterWriter are non-null, else both are null
-      void DeclareField(string name, bool isStatic, bool isConst, Type type, Bpl.IToken tok, string rhs);
+      void DeclareField(string name, List<TypeParameter> targs, bool isStatic, bool isConst, Type type, Bpl.IToken tok, string rhs);
```

Comment:
This seems reasonable, but I'm curious why it's necessary.

Reply:
If you have a `static  const` field in a Dafny class that takes a parameter, e.g., something like this:
```
class Vector<T>
{
    static const DEFAULT_SIZE :uint32 := 16
    var buffer:array<T>;
}
```
then when we generate the `.cpp` file, we need to separately instantiate the static class member, which in C++, requires including the template arguments again; i.e., we generate:
```
template <typename T> uint32 Vector<T>::DEFAULT__SIZE = (uint32)16;
```

Reply:
Makes sense. Might it be better to pass the declaring class along instead, since that's where the type parameter is declared in the first place?

Reply:
If that's the preference, I can swap to that style.  In several places, it seemed the preference was to pass components, rather than the original object.  Passing the object does seem like it makes the interface a bit more "future proof" if someone later wants a different element of the object.

Reply:
Okay, I changed it to pass in a TopLevelDecl instead of the list of TypeParameters.  That seems to be compatible with all of the call sites.